temporal-sdk/src/main/java/io/temporal/internal/worker/LocalActivityExecutionContext.java

```diff
@@ -154,6 +161,14 @@ public boolean callback(LocalActivityResult result) {
     if (scheduleToCloseFuture != null) {
       scheduleToCloseFuture.cancel(false);
     }
+    SlotReleaseReason reason = SlotReleaseReason.taskComplete();
+    if (result.getProcessingError() != null) {
+      reason = SlotReleaseReason.error(new Exception(result.getProcessingError().getThrowable()));
+    }
+    // Permit can be null in the event of a timeout while waiting on a permit
+    if (permit != null) {
+      slotSupplier.releaseSlot(reason, permit);
```

Comment:
Didn't we say we would move the release out of here?

Reply:
Yes, good point. And move it instead to `handle`, which is more like the normal task types?

Reply:
I guess the main issue is that this gets referenced from like a bajillion places, so it's not obvious that centralizing it is going to catch all those... but I can see what happens.

Reply:
It also looks like your release the slot again  in `handleResult`? Maybe it is being release twice.  The slot acquire/release. `handle` is the right spot IMO, currently the SDK takes and releases slots after handle.

Reply:
At a high level the slot  acquire/release should be tied to the processing of the task, not the reporting of the result.

Reply:
:+1: done